syntax = "proto3";

package datarequests;


message OrderPrice {
  float base_order_price = 1;
  float coin_coef = 2;
  float coin_bonus_amount = 3;
  float final_order_price = 4;
}

message ExecuterProfile {
  string id = 1;
  repeated string tags = 2;
  float rating = 3;
}

message GetOrderInfoRequest {
  string order_id = 1;
  string executer_id = 2;
}

message GetOrderInfoResponse {
  OrderPrice order_price = 1;
  string zone_display_name = 2;
  ExecuterProfile executor_profile = 3;
}

service DataRequestsService {
  rpc GetOrderInfo(GetOrderInfoRequest) returns (GetOrderInfoResponse);
}
